DSharpPlus.Commands/CommandExecutor.cs

```diff
@@ -74,23 +74,49 @@ await context.Extension._commandErrored.InvokeAsync(context.Extension, new Comma
 
         if (checks.Count != 0)
         {
-            Dictionary<ContextCheckAttribute, Exception> failedChecks = [];
+            List<(ContextCheckAttribute data, string error)> failedChecks = [];
 
-            // Reverse foreach so we execute the top-most command's checks first.
-            for (int i = checks.Count - 1; i >= 0; i--)
+            // first, execute all unconditional checks
+            foreach (ContextCheckMapEntry entry in context.Extension.Checks.Where(x => x.AttributeType == typeof(UnconditionalCheckAttribute)))
             {
                 try
                 {
-                    if (!await checks[i].ExecuteCheckAsync(context))
+                    object check = ActivatorUtilities.CreateInstance(context.Extension.ServiceProvider, entry.CheckType);
+                    string? result = await entry.ExecuteCheckAsync(check, new UnconditionalCheckAttribute(), context);
+
+                    if (result is not null)
                     {
-                        failedChecks.Add(checks[i], new Exception("placeholder exception until checks are redone"));
+                        failedChecks.Add((new UnconditionalCheckAttribute(), result));
                         continue;
                     }
                 }
-                // try/catch blocks are free until they catch
                 catch (Exception error)
                 {
-                    failedChecks.Add(checks[i], error);
+                    failedChecks.Add((new UnconditionalCheckAttribute(), $"{error}: {error.Message}\n{error.StackTrace}"));
```

Comment:
`error.ToString()`?

Reply:
> https://learn.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=net-8.0#remarks

`Environment.StackTrace` loses information that `Exception.StackTrace` provides, specifically on async methods. Keeping as is.